cosmos/converter.py

```diff
@@ -155,6 +161,10 @@ def __init__(
             "profile_config": profile_config,
             "emit_datasets": emit_datasets,
         }
+
+        if execution_mode == ExecutionMode.VIRTUALENV and execution_config.virtualenv_dir is not None:
```

Comment:
Are we creating the path if it doesn't exist? It may be worth checking if the directory path exists.

Reply:
Great suggestion, will add :)